### PR TITLE
null check in nested key resolution

### PR DESCRIPTION
**Pull request by cliewpaypal**
**What**: 

**Why**: https://github.com/kentcdodds/match-sorter/issues/31

**How**:

**Checklist**:

- [ ] Documentation
- [ ] Tests
- [ ] Ready to be merged 
- [ ] Added myself to contributors table

### CODE REVIEW

**Comment by kentcdodds** [comment]
The build is failing because we're not getting 100% code coverage with these changes. You could probably just add an item to [this list](https://github.com/kentcdodds/match-sorter/blob/8a2072591d310f8010a381504a8bf298d9391adf/src/__tests__/index.js#L108) where `name: null` and that should cover the new code :+1:

File: src/__tests__/index.js [hunk 1]
```diff
@@ -105,7 +105,7 @@ const tests = {
   },
   'can handle objected with nested keys': {
     input: [
-      [{name: {first: 'baz'}}, {name: {first: 'bat'}}, {name: {first: 'foo'}}],
+      [{name: {first: 'baz'}}, {name: {first: 'bat'}}, {}],
```
Comment (kentcdodds) [inline src/__tests__/index.js]: Could you actually add another item to the array? This test is also ensuring that we filter out the foo item.

File: src/index.js [hunk 2]
```diff
@@ -251,7 +251,7 @@ function getItemValues(item, key) {
     // handle nested keys
     value = key
       .split('.')
-      .reduce((itemObj, nestedKey) => itemObj[nestedKey], item)
+      .reduce((itemObj, nestedKey) => itemObj && itemObj[nestedKey] ? itemObj[nestedKey] : null, item)
```
Comment (kentcdodds) [inline src/index.js]: The second part of the `&&` is unnecessary here

File: src/__tests__/index.js (hunk 1 above)
Comment (cliewpaypal) [reply src/__tests__/index.js]: oh right gotcha.

**Comment by kentcdodds** [comment]
Thanks a bunch!

**Pull request merged**